Installing blueprint v5-alpha: Unable to resolve dependency tree

<h3>Current Behavior:</h3>

When I run `npm i @blueprintjs/core@5.0.0-alpha.6 @blueprintjs/icons@5.0.0-alpha.4`, on my project which uses react 18, I get:

<code>npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: project@1.0.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8 || 17" from @blueprintjs/icons@5.0.0-alpha.4
npm ERR! node_modules/@blueprintjs/icons
npm ERR!   @blueprintjs/icons@"5.0.0-alpha.4" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
</code>

<h3>Expected behavior:</h3>
I expect npm to upgrade my package without errors

<h3>Possible solution</h3>

Why peerDependencies section in the `@blueprintjs/icons@5.0.0-alpha.4` package don't support 18 version of react? Maybe this will solve the problem?

--- a/packages/icons/package.json
+++ b/packages/icons/package.json
@@ -47,8 +47,14 @@
         "tslib": "~2.5.0"
     },
     "peerDependencies": {
-        "react": "^16.8 || 17",
-        "react-dom": "^16.8 || 17"
+        "@types/react": "^16.14.32 || 17 || 18",
+        "react": "^16.8 || 17 || 18",
+        "react-dom": "^16.8 || 17 || 18"
+    },
+    "peerDependenciesMeta": {
+        "@types/react": {
+            "optional": true
+        }
     },
     "devDependencies": {
         "@blueprintjs/node-build-scripts": "^8.0.0-alpha.3",

